WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -177,12 +178,16 @@ public static void addMedia(Activity activity) {
     }
 
     public static void viewMyProfile(Context context) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
+
         Intent intent = new Intent(context, MyProfileActivity.class);
         AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.OPENED_MY_PROFILE);
         slideInFromRight(context, intent);
     }
 
     public static void viewAccountSettings(Activity activity) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
```

Comment:
I kind of disagree with that. Moving to the activities will mean that if the activity is restarted, we'll be fetching the details again and keeping the fetch here prevents that. I'd also argue that, it's not that different than adding the analytics call here. I'd consider this as more of a central dispatcher than a helper class which just creates the intents.

In an ideal case, the lower layers would decide when to do fetching with the rules we specify (unless it's forced by the user). However, with the current network structure, that's not easy to do.

I can still move it to the Activity, if you feel strongly about it, but I thought I'd at least share my point of view. Cheers!